include/vcpkg/base/files.h

```diff
@@ -237,6 +237,9 @@ namespace vcpkg
         virtual std::vector<Path> get_regular_files_recursive(const Path& dir, std::error_code& ec) const = 0;
         std::vector<Path> get_regular_files_recursive(const Path& dir, LineInfo li) const;
 
+        virtual std::vector<Path> get_regular_files_recursive_proximate(const Path& dir, std::error_code& ec) const = 0;
```

Comment:
Should we denote anywhere herein that it's lexically proximate? I don't think we're resolving symlinks anywhere.